lfs.c

```diff
@@ -3064,11 +3064,26 @@ static lfs_soff_t lfs_file_rawseek(lfs_t *lfs, lfs_file_t *file,
     // find new pos
     lfs_off_t npos = file->pos;
     if (whence == LFS_SEEK_SET) {
-        npos = off;
+        if (off < 0) {
+            return LFS_ERR_INVAL;
+        }
+        else {
+            npos = off;
+        }
     } else if (whence == LFS_SEEK_CUR) {
-        npos = file->pos + off;
+        if ((lfs_soff_t)file->pos + off < 0) {
+            return LFS_ERR_INVAL;
+        }
+        else {
+            npos = file->pos + off;
+        }
     } else if (whence == LFS_SEEK_END) {
-        npos = lfs_file_rawsize(lfs, file) + off;
+        if (lfs_file_rawsize(lfs, file) + off < 0) {
```

Comment:
We should avoid calling `lfs_file_rawsize` multiple times, since it is actually a bit complex.